daphne/src/testing.rs

```diff
@@ -701,20 +734,22 @@ where
         let leader_state_store = leader_state_store_mutex_guard.deref_mut();
 
         // Construct a new Collect URI for this CollectReq.
-        let collect_id = Id(rng.gen());
+        let collect_id = if let Some(cid) = collect_job_id {
+            cid.clone()
+        } else {
+            CollectionJobId(rng.gen())
+        };
```

Comment:
* More idiomatic
* Worth commenting that the caller doesn't specify a collection ID in draft02.
```suggestion
        // draft02: No collection ID is indicated in the request path, so we generate one ourselves.
        let collect_id = collect_job_id.map_or(CollectionJobId(rng.gen(), |cid| cid.clone());
```

Reply:
This ended up as:

let collect_id = collect_job_id
            .as_ref()
            .map_or_else(|| CollectionJobId(rng.gen()), |cid| cid.clone());

to deal with avoiding a move of collect_job_id and also to not evaluate rng.gen() if we didn't need to